engines/hdb/file-manager.cpp

```diff
@@ -0,0 +1,221 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+#include "common/debug.h"
+#include "common/file.h"
+#include "common/memstream.h"
+#include "common/zlib.h"
+
+#include "hdb/hdb.h"
+#include "hdb/file-manager.h"
+
+namespace HDB {
+
+FileMan::FileMan() {
+	_mpcFile = new Common::File;
+}
+
+FileMan::~FileMan() {
+	delete _mpcFile;
+	for (uint i = 0; i < _dir.size(); i++)
+		delete _dir[i];
+}
+
+bool FileMan::openMPC(const Common::String &filename) {
+	uint32 offset;
+
+	if (!_mpcFile->open(filename)) {
+		error("FileMan::openMPC(): Error reading the MSD/MPC file %s", filename.c_str());
+		return false;
+	}
+
+	_dataHeader.id = _mpcFile->readUint32BE();
+
+	if (_dataHeader.id == MKTAG('M', 'P', 'C', 'C')) {
+		debug("COMPRESSED MPC FILE");
+		return false;
+	} else if (_dataHeader.id == MKTAG('M', 'P', 'C', 'U')) {
+		// we're fine
+	} else if (_dataHeader.id == MKTAG('M', 'S', 'D', 'C')) {
+		// we're fine
+	} else if (_dataHeader.id == MKTAG('M', 'S', 'D', 'U')) {
+		debug("UNCOMPRESSED MSD FILE");
```

Comment:
The problem here is that in the original sources, all the four headers MPCU, MPCC, MSDU, MSDC have been defined even though the data files that we have only match two of them. Because all of the four were mentioned, we have put all four of them and the missing ones have stubbed code.